docs/src/tutorial.md

```diff
@@ -0,0 +1,283 @@
+# JET tutorial
+JET is a package that leverages the Julia compiler's inference to check user code for type instability and type errors.
+This tutorial will demonstrate how to use JET effectively.
+It presupposes the reader has working knowledge of Julia, and understands Julian concepts such as "dynamic dispatch" and "type stability".
```

Comment:
Needs links on where to learn what those are.

Reply:
Done